pycobertura/cobertura.py

```diff
@@ -77,32 +77,50 @@ def _load_from_file(self, report_file):
     def _load_from_string(self, s):
         return ET.fromstring(s)
 
+    @memoize
+    def _get_class_elements_by_filename(self, filename):
+        syntax = f"./packages//class[@filename='{filename}']"
+        classes = []
+        for c in self.xml.xpath(syntax):
+            classes.append(c)
+        return classes
+
     @memoize
     def _get_class_element_by_filename(self, filename):
-        syntax = "./packages//class[@filename='%s'][1]" % (filename)
+        syntax = f"./packages//class[@filename='{filename}'][1]"
         return self.xml.xpath(syntax)[0]
 
     @memoize
     def _get_lines_by_filename(self, filename):
-        el = self._get_class_element_by_filename(filename)
-        return el.xpath("./lines/line")
+        classElements = self._get_class_elements_by_filename(filename)
+        return [
+            line
+            for classElement in classElements
+            for line in classElement.xpath("./lines/line")
+        ]
 
     @property
     def version(self):
         """Return the version number of the coverage report."""
-        return self.xml.attrib["version"]
+        return self.xml.get("version")
 
     def line_rate(self, filename=None):
         """
         Return the global line rate of the coverage report. If the
         `filename` file is given, return the line rate of the file.
         """
+        lineRate = 0
         if filename is None:
-            el = self.xml
+            lineRate = float(self.xml.get("line-rate"))
```

Comment:
Can we early return here?

```suggestion
            return float(self.xml.get("line-rate"))
```

... and remove the `else` statement below.

Reply:
done in https://github.com/aconrad/pycobertura/pull/132/commits/45e1b7b103bd59c9ff4756375a69fbcd479d0608